odl/solvers/functional/default_functionals.py

```diff
@@ -323,15 +354,28 @@ def convex_conj(self):
         return IndicatorGroupL1UnitBall(self.domain, exponent=conj_exp)
 
     def __repr__(self):
-        """Return ``repr(self)``."""
-        return '{}({!r}, exponent={})'.format(self.__class__.__name__,
-                                              self.domain,
-                                              self.pointwise_norm.exponent)
+        """Return ``repr(self)``.
+
+        Examples
+        --------
+        >>> pspace = odl.ProductSpace(odl.rn(2), 2)
+        >>> l1_2_norm = odl.solvers.GroupL1Norm(pspace, exponent=2)
+        >>> l1_2_norm
+        GroupL1Norm(ProductSpace(rn(2), 2), exponent=2.0)
+        """
+        posargs = [self.domain]
+        optargs = [('exponent', self.pointwise_norm.exponent, None)]
+        with npy_printoptions(precision=REPR_PRECISION):
+            inner_parts = signature_string_parts(posargs, optargs)
+        return repr_string(self.__class__.__name__, inner_parts,
+                           allow_mixed_seps=False)
 
 
 class IndicatorGroupL1UnitBall(Functional):
 
-    """The convex conjugate to the mixed L1--Lp norm on `ProductSpace`.
+    """Indicator functional of the unit ball in the Group-L1 norm.
```

Comment:
For all of these, we should perhaps be more clear in what we mean by "indicator function" since this has different meanings in different contexts.